src/Raven.Server/Documents/Indexes/Static/IndexCompilationCache.cs

```diff
@@ -50,7 +50,12 @@ public static StaticIndexBase GetIndexInstance(IndexDefinition definition, Raven
 
             try
             {
-                return result.Value;
+                var index = result.Value;
```

Comment:
The cache is used for reusing the same instance of an index and this is why it is in the cache other than that the index is unique since it has a state unlike the c# ones

Reply:
Can we move this from static to the database level?

Reply:
no we can't and refactoring how indexing is cached is out of the scope of this PR lets focus on memory leaks and not further expend this PR

Reply:
I just created one: https://issues.hibernatingrhinos.com/issue/RavenDB-13819 separating the work will allow us to divide and conquer.